pkg/oidc/provider/error/error_test.go

```diff
@@ -0,0 +1,39 @@
+package error
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWriteError(t *testing.T) {
+	const (
+		errorMessage     = "message"
+		errorDescription = "description"
+		errorCode        = 401
+	)
+	rec := httptest.NewRecorder()
+
+	WriteError(errorMessage, errorDescription, errorCode, rec)
+
+	assert.Equal(t, errorCode, rec.Code)
+	assert.Equal(t, `{"error":"`+errorMessage+`","error_description":"`+errorDescription+`"}`, strings.TrimSpace(rec.Body.String()))
```

Comment:
It should not happen but `github.com/stretchr/testify/assert` has a [`assert#JSONEq`](https://pkg.go.dev/github.com/stretchr/testify/assert#JSONEq) to compare JSON strings. I guess it handles the possible order of the keys (that it's not granted).